include/tfm/tfm_ioctl_api.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef TFM_IOCTL_API_H__
+#define TFM_IOCTL_API_H__
+
+#include <limits.h>
+#include <stdint.h>
+#include <tfm_api.h>
+#include <tfm_platform_api.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum tfm_platform_ioctl_reqest_types_t {
+    TFM_PLATFORM_IOCTL_READ_SERVICE
+};
+
+/*!
+ * \struct tfm_read_service_args_t
+ *
+ * \brief Argument list for each platform read service
+ */
+struct tfm_read_service_args_t {
+	void *destination;
+	uint32_t addr;
+	size_t len;
+};
+
+/*!
+ * \struct tfm_read_service_out_t
+ *
+ * \brief Output list for each read platform service
+ */
+struct tfm_read_service_out_t {
+	uint32_t result;
+};
+
+/*!
+ * \brief Perform a read operation.
+ *
+ * \param[in]  pin_num       Pin number of the selected pin
+ * \param[in]  direction     Direction of the pin: 0 for input, 1 for output
+ * \param[out] result        Return error value
+ *
+ * \return Returns values as specified by the \ref tfm_platform_err_t
+ */
+enum tfm_platform_err_t tfm_platform_read(void *destination, uint32_t addr,
```

Comment:
Should this be named "read_memory", or "read_mem"?
I guess this is used only for memory-mapped "readings"

Reply:
changed to "mem_read" as this is the most used variant.